### PR TITLE
Add support for bpftrace

### PR DESCRIPTION
**Pull request by bcmyers**
I think I was able to write a pretty efficient parser for bpftrace input. Let me know what you think!

### CODE REVIEW

**Comment by bcmyers** [comment]
Shoot. I just wanted to push you the last commit ("port stackcollapse-bpftrace.pl"). I'm fairly new to git; so how do I fix things so that that's the only commit that shows up in this PR.

**Comment by jonhoo** [comment]
Try this:
```console
$ git fetch origin
$ git rebase origin/master HEAD
```
To verify, check that you only get your one commit with:
```console
$ git log origin/master..
```
In general, I would recommend to always make changes in a new branch forked from master:
```console
$ git fetch origin
$ git branch add-bpftrace origin/master
$ git checkout add-bpftrace
```
That way you don't end up building on top of a stale copy of the code. You can do that in this case too by "cherry-picking" the one commit you've already made onto the `add-bpftrace` branch:
```console
$ git cherry-pick 0ada8f4f898352e732f6145854e82d30d2997b8f
```

File: src/collapse/bpftrace.rs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+        let root_dir = env::var("CARGO_MANIFEST_DIR").unwrap();
+        let data_dir = Path::new(&root_dir)
+            .join("tests")
+            .join("data")
+            .join("bpftrace");
+        let perl = Path::new(&root_dir)
+            .join("flamegraph")
+            .join("stackcollapse-bpftrace.pl");
+
+        for i in 1..NUMBER_OF_TEST_INPUTS + 1 {
```
Comment (jonhoo) [inline src/collapse/bpftrace.rs]: I think I would probably instead just [walk the files in the directory](https://doc.rust-lang.org/std/fs/fn.read_dir.html).

**Comment by jonhoo** [comment]
Note that you _may_ have to open a new pull request (and close this one) you start working from a new branch.

File: src/bin/collapse-bpftrace.rs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+const READER_CAPACITY: usize = 128 * 1024;
+
+#[derive(Debug, StructOpt)]
+#[structopt(name = "inferno-collapse-bpftrace", author = "")]
+struct Opt {
+    /// bpftrace script output file, or STDIN if not specified
+    infile: Option<PathBuf>,
+}
+
+fn main() -> io::Result<()> {
```
Comment (jonhoo) [inline src/bin/collapse-bpftrace.rs]: I think it would be good if we could avoid all the duplication between this and the `collapse-perf.rs`..

**Comment by bcmyers** [comment]
I'll close this one and start over. Thanks!

**Pull request closed without merging**